walkers/src/download.rs

```diff
@@ -72,19 +108,62 @@ where
     // Keep outside the loop to reuse it as much as possible.
     let client = http_client(http_options);
 
+    let mut ongoing_downloads = Vec::<_>::new();
+
     loop {
-        let request = request_rx.next().await.ok_or(())?;
-        let url = source.tile_url(request);
+        let request = request_rx.next();
+        let below_concurrent_downloads_limit = ongoing_downloads.len() < 2;
```

Comment:
Is this really limit the connections?
In my world, one Client == one connection.
For http2 (tile.openstreetmaps.org supports multiplexing) this looks like obstacle.

Maybe better to have two client workers (to restrict number of connections) and limit download queue by any reasonable value (like a half of number of tiles on display)?

Reply:
> I have a gut feeling that it does though.

I think different because http/2 streams not require system resources (like port), and they not leave many zombie-connections (in state CLOSE_WAIT).

I think we can forget about http/1 because osm.org uses http2. This can be checked using this site: https://tools.keycdn.com/http2-test or curl: 
```
curl -v -v -v https://tile.openstreetmap.org/18/187598/121559.png -o /dev/null 2>&1 | grep -i http2
* Using HTTP2, server supports multiplexing
```

About testing many connections in current way:
TCP sockets take to change status: so far first client will idle in state CLOSING/FIN_WAIT/CLOSE_WAIT, next client will download the tile. Next, the roles will change and server always assume that more are connected to him if I understand correctly how TCP works.

Reply:
![image](https://github.com/podusowski/walkers/assets/324098/6239f98c-64f0-44ed-9a00-ce93b0fa526d)
`watch -n 1 "bash -c 'netstat -aptun| grep target'"
 says what only one stable connection is used in `multi-download` branch

Reply:
OK, I guess I need to do more thinking on this. In the meanwhile, I have checked how Leaflet does this. I've disabled both HTTP/2 and HTTP/3.

![image](https://github.com/podusowski/walkers/assets/2226778/037004bf-9d1c-4abc-8225-28bfe3e58f99)

Reply:
this looks like 6 connections in parallel

Reply:
Seems consistent with the statement over here: https://github.com/seanmonstar/reqwest/issues/1517#issue-1190144771

Well then, the approach I've taken is completely wrong. This limit needs to be enforced by reqwest/hyper, not here.

Reply:
similar result: 6 permanent connections to tile.osm.org (`151.101.*.*`) and 6 parallel downloads

Reply:
Setting the limit to 6 seems to give pretty pleasing experience. If `reqwest` can do http/2 (which I believe it can), it should use a single connection anyway, so it should meet the OSM rules.